scheduler/filter/expr.go

```diff
@@ -74,17 +74,24 @@ func parseExprs(key string, env []string) ([]expr, error) {
 
 func (e *expr) Match(whats ...string) bool {
 	var (
-		pattern string
-		match   bool
-		err     error
+		pattern    string
+		match      bool
+		err        error
+		startIndex int
 	)
 
-	if e.value[0] == '/' && e.value[len(e.value)-1] == '/' {
+	if e.IsSoft() {
```

Comment:
Thats actaully nice thing to do we can have IsSoft bool in type expr and trim **~** int the expression is a better idea . I'll make the necessary changes and update the PR